src/shapes/text.class.js

```diff
@@ -1570,11 +1580,23 @@
    * Returns fabric.Text instance from an object representation
    * @static
    * @memberOf fabric.Text
-   * @param {Object} object Object to create an instance from
+   * @param {Object} object plain js Object to create an instance from
    * @param {Function} [callback] Callback to invoke when an fabric.Text instance is created
    */
   fabric.Text.fromObject = function(object, callback) {
-    return fabric.Object._fromObject('Text', object, callback, 'text');
+    var objectCopy = clone(object), path = object.path;
+    delete objectCopy.path;
```

Comment:
@melchiar when i made that clone comment, i meant this:
we clone the input object, so that we can safely delete the path from our copy without mutating the input.

Reply:
got it, i wasn't sure.